### DISCUSSION TITLE
brew cleanup removing files needed by installed formulae

### DISCUSSION THREAD

**Discussion by pulberg** [Homebrew/discussions#1454] [Everyday usage] [answered]
I am using a formulae called jenv which lets you specify the use of multiple java versions and where\how to use them.  When brew cleanup runs, it is removing files used by the app - 

Removing: /usr/local/Cellar/jenv/plugins... (64B)
Removing: /usr/local/Cellar/jenv/shims... (54 files, 3.0KB)
Removing: /usr/local/Cellar/jenv/versions... (152 files, 97.9KB)

This results in a very broken jenv - 

jenv
-bash: /usr/local/Cellar/jenv/shims/jenv: No such file or directory

I have no idea why this is happening or where the issue is, brew or the formulae?

**Comment by carlocab** [comment]
Shouldn't `jenv` be putting those directories in `~/.jenv`? How are they ending up in `/usr/local/Cellar/jenv`?

**Reply by pulberg** [reply]
No idea, i'm just installing like I do with all of the other formulae I use. `brew install jenv`

**Comment by gromgit** [comment, marked as answer]
Did you set the `JENV_ROOT` environment variable to `/usr/local/Cellar/jenv`? That would force `jenv` to put everything right where it did, and since the only thing Homebrew expects under `/usr/local/Cellar/<formula>` is a `<version>` subdirectory, `brew cleanup` naturally wipes your existing setup.

Unset `JENV_ROOT`, then `eval "(/usr/local/bin/jenv init -)"` again to set up a new hierarchy under `~/.jenv` as intended. You may have to add `${HOME}/.jenv/bin` to your `PATH` as well, as directed in the `jenv` caveats.

**Reply by pulberg** [reply]
That was it, thanks for getting me in the right direction.
